ota_university/ctf_framework/models.py

```diff
@@ -0,0 +1,111 @@
+from django.db import models
+from django.contrib.auth.models import User
+from django_slack_oauth.models import SlackUser
+from django.core.exceptions import ObjectDoesNotExist
+
+
+class CTFSlackUser(SlackUser):
+    """
+    Extending the django oauth SlackUser to support display names
+    """
+    display_name = models.CharField(max_length=100)
+
+
+class Title(models.Model):
+    """
+    Awardable User Titles
+    """
+    title = models.CharField(max_length=30)
+
+    def __str__(self):
+        return self.title
+
+
+class ChallengeCategory(models.Model):
+    """
+    CTF Challenge Categories that can be dynamically added
+    Objects cannot be deleted if a Challenge references them
+    """
+
+    class Meta:
+        # For Displaying in Admin Panel
+        verbose_name = "Challenge Categorie"
```

Comment:
Ideally, we should use pluralization : https://dustindavis.me/django-pluralize-outside-templates/
The change here should be simple and avoids confusion where people would consider this a typo.